### ISSUE TITLE
Docker Containerization

### ISSUE THREAD

**Issue by SiqingYu** [gpodder/mygpo#355]
Moved from https://github.com/gpodder/mygpo/issues/262

**Comment by SiqingYu** [comment]
Your advice would be appreciated. :smile:
@stefankoegl @JeanFred @morgenroth @fabolhak @elelay

**Comment by stefankoegl** [comment]
I think I have a solid understanding of what containerization is, but I have never worked with kubernetes myself. We should keep in mind that we don't make it too complicated to get a test system up and running, to ensure that new contributors can join easily.

More generally, I think gpodder.net's architecture allows for it to be split up into micro services if that's what you're going for. I'm happy to discuss that further (eg on Slack).

**Comment by ryoung29** [comment]
I can help with this as well

**Comment by SiqingYu** [comment]
@ryoung29 Thanks. There is a WIP PR that you can follow. https://github.com/gpodder/mygpo/pull/32

**Comment by zb3** [comment]
I'm currently trying to set up a development environment using docker compose (I'm learning and my goal is to do `docker-compose up` and have it running).

I saw issue #262 and looked into PR #32, and it appears to me that containerization here is ultimately concerned with the production use (as opposed to #262). The development setup I'm currently trying to create has:
* `DEBUG=true` and using `manage.py runserver` (exceptions are visible and static files are served)
* source directory mounted as a volume so no need to rebuild the image to test changes
* test dependencies installed
* remote debugging support

Since my points above seem to conflict with production use, maybe there could be a separate docker image and compose file for the development setup? If so, I might be able to help with that once I get that running properly.

**Comment by oliverkane** [comment]
Hi there. I found this Issue while I was on my way to dockerize this myself since I didn't see anything related to that in the master branch. Glad to see this is in progress.  What's the best way for me to connect to folks and see where I could best contribute to this effort?  My end goal is to get this into k8s, and I'm pretty familiar with dockerizing apps, though I'm a bit poor at python.

**Comment by PascalNoisette** [comment]
Hi,

I came accross the april 2021 fork of  [@maxolasersquad](https://github.com/maxolasersquad/mygpo-docker)  It is running quite well out of the box and I think it solve #32 and [partially this issue](#) (not the k8s part, it's implemented with docker-compose).

@zb3, I am running behind a nginx reverse proxy with DEBUG: "True" the image: [maxolasersquad/mygpo](https://hub.docker.com/r/maxolasersquad/mygpo ) so you might be interested as well.  (celery must be restarted *after* postgres though and after migrate, otherwise I get some Relation "django_celery_beat_periodictask" does not exist).

Regards,

**Comment by nagyv** [comment]
I created a dockerized version that I use in Kubernetes. It's available at https://gitlab.com/nagyv/gpodder

Please, pick any of my code and contribute it here.

**Comment by almereyda** [comment]
@nagyv FYI the repository https://gitlab.com/nagyv/gpodder is 404 nowadays. I suspect it is superseded by

- #809

**Comment by nagyv** [comment]
@almereyda Correct. the PR includes minimal docs too and it would add all the changes I made to the main `mygpo` repo.

**Comment by albertopasqualetto** [comment]
Is there any update or ETA?

**Comment by Ryochan7** [comment]
Closest thing you will get would be at https://codeberg.org/atomdmac/mygpo-docker/. It allows running the mygpo project and related dependencies in Docker.
